packages/styled-system/src/types.ts

```diff
@@ -41,7 +41,7 @@ type PseudoKeys = keyof CSS.Pseudos | keyof Pseudos
 
 type PseudoSelectorDefinition<D> = D | RecursivePseudo<D>
 
-type RecursivePseudo<D> = {
+export type RecursivePseudo<D> = {
```

Comment:
@TimKolberger I had to add this export, otherwise TS was throwing this error in `resolve-responsive-prop-styles.ts`:
```
src/resolve-responsive-prop-styles.ts(21,17): error TS4058: Return type of exported function has or is using name 'RecursivePseudo' from external module "/Users/jozefmikulas/dev/chakra-ui/packages/styled-system/dist/types/types" but cannot be named.
```

I have never seen that 🤷‍♂️ Any other idea?

Reply:
`RecursivePseudo` is part of `StyleObjectOrFn` and the build process is generating the .d.ts files for each library. If it appears there, it needs to be exported from its origin. Thats fine 👍